mlir/test/Conversion/ArithToEmitC/arith-to-emitc.mlir

```diff
@@ -355,6 +361,27 @@ func.func @arith_cmpi_predicates(%arg0: i32, %arg1: i32) {
   return
 }
 
+func.func @arith_cmpi_index(%arg0: i32, %arg1: i32) -> i1 {
+  // CHECK-LABEL: arith_cmpi_index
+
+  // CHECK: %[[Cst0:.*]] = emitc.cast %{{.*}} : {{.*}} to !emitc.size_t
```

Comment:
Good catch, there's a double cast happening here (i32 -> ssize_t -> size_t):
* i32 -> ssize_t is per arith.index_cast which is signed,
* ssize_t -> size_t is to match `index` being converted into unsigned `size_t` by the type converter.

I don't want to match the double cast here, since this might change in the future; plus, I want to check cmpi's behavior in this test and avoid adding noise from other ops.